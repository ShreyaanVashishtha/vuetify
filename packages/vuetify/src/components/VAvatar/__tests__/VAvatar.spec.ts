// Components
import { VAvatar } from '../'

// Utilities
import { h } from 'vue'
import { createVuetify } from '@/framework'
import { mount } from '@vue/test-utils'

describe('VAvatar', () => {
  const vuetify = createVuetify()
  const mountFunction = (options = {}) => mount(VAvatar, {
    ...options,
    global: {
      plugins: [vuetify],
    },
  })

  it('should have an v-avatar class', () => {
    const wrapper = mountFunction()

    expect(wrapper.classes()).toContain('v-avatar')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should respect size prop', () => {
    const wrapper = mountFunction({
      props: {
        size: 'large',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should respect rounded prop', () => {
    const wrapper = mountFunction({
      props: {
        rounded: 0,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should respect color prop', () => {
    const wrapper = mountFunction({
      props: {
        color: 'error',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should respect tag prop', () => {
    const wrapper = mountFunction({
      props: {
        tag: 'span',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render default slot content', () => {
    const wrapper = mountFunction({
      slots: {
        default: () => h('span', 'foo'),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
