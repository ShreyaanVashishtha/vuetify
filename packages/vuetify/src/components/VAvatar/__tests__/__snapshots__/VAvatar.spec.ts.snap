// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VAvatar should have an v-avatar class 1`] = `
<div class="v-avatar v-avatar--size-default">
</div>
`;

exports[`VAvatar should render default slot content 1`] = `
<div class="v-avatar v-avatar--size-default">
  <span>
    foo
  </span>
</div>
`;

exports[`VAvatar should respect color prop 1`] = `
<div class="v-avatar bg-error v-avatar--size-default">
</div>
`;

exports[`VAvatar should respect rounded prop 1`] = `
<div class="v-avatar rounded-0 v-avatar--size-default">
</div>
`;

exports[`VAvatar should respect size prop 1`] = `
<div class="v-avatar v-avatar--size-large">
</div>
`;

exports[`VAvatar should respect tag prop 1`] = `
<span class="v-avatar v-avatar--size-default">
</span>
`;
